src/components/NotFound.astro

```diff
@@ -132,7 +140,7 @@ const dictionary = Object.fromEntries(
 				const dictionary = JSON.parse(this.dataset.dictionary);
 
 				// Get current language from the URL path.
-				const langCodeMatch = location.pathname.match(/\/([a-z]{2}-?[a-z]{0,2})\//);
+				const langCodeMatch = location.pathname.match(/\/([a-z]{2}(-[a-z]{0-2})?)\//);
```

Comment:
Thanks for explaining! I’d missed the original issue about this, so didn’t have that context — sorry if I seemed a bit dense 😅 

So thinking about this a little more, this issue still persists if the route is `/aa` or `/zz` or something similar that matches our regex but is not actually a language. We may want to move all the mutations behind a check that `translations` exists. This would assure us that the match was a valid language and only update the link, `lang` attribute, and text if so.

Reply:
Yep! Sounds great. Pushed a new version that builds a regex from our available languages, so I believe `lang` should already default to `en` if a known language isn't found, therefore `translations` should always exist.